yarn-project/end-to-end/src/e2e_synching.test.ts

```diff
@@ -0,0 +1,428 @@
+        const sender = this.wallets[i].getAddress();
+        const recipient = this.wallets[(i + 1) % this.txCount].getAddress();
+        const tk = await TokenContract.at(this.token.address, this.wallets[i]);
+        txs.push(tk.methods.transfer_public(sender, recipient, 1n, 0).send());
+      }
+      return txs;
+    } else if (this.txComplexity == TxComplexity.MaxDiff) {
+      // This one is slightly more painful. We need to setup a new contract that writes
+      // a metric ton of state changes.
+
+      const txs = [];
+      for (let i = 0; i < this.txCount; i++) {
+        const batch = new BatchCall(this.wallets[i], [
+          this.spam.methods.private_spam(this.seed, 16, false).request(),
+          this.spam.methods.private_spam(this.seed + 16n, 16, false).request(),
+          this.spam.methods.private_spam(this.seed + 32n, 16, false).request(),
+          this.spam.methods.private_spam(this.seed + 48n, 15, true).request(),
+        ]);
+
+        this.seed += 100n;
+        txs.push(batch.send());
+      }
+      return txs;
+    } else {
+      throw new Error('Incorrect tx complexity');
+    }
+  }
+
+  async writeBlocks(blocks: L2Block[]) {
+    await this.writeJson(`blocks`, { blocks: blocks.map(block => block.toString()) });
+  }
+
+  loadBlocks() {
+    const json = this.loadJson(`blocks`);
+    return (json['blocks'] as string[]).map(b => L2Block.fromString(b));
+  }
+
+  async writeStats(content: Record<string, string | number>) {
+    await this.writeJson(`stats`, {
+      description: this.description(),
+      ...content,
+    });
+  }
+
+  numberOfBlocksStored() {
+    const files = fs.readdirSync(this.dir());
+    return files.filter(file => file.startsWith('block_')).length;
+  }
+
+  private async writeJson(fileName: string, toSave: Record<string, string[] | string | number>) {
+    const dir = this.dir();
+    await fs.promises.mkdir(dir, { recursive: true });
+
+    const path = `${dir}/${fileName}.json`;
+    const output = JSON.stringify(toSave, null, 2);
+    fs.writeFileSync(path, output, 'utf8');
+  }
+
+  private loadJson(fileName: string): Record<string, string[] | string | number> {
+    const path = `${this.dir()}/${fileName}.json`;
+    const data = fs.readFileSync(path, 'utf8');
+    return JSON.parse(data);
+  }
+
+  private dir(): string {
+    return `./src/fixtures/synching_blocks/${this.name()}`;
+  }
+}
+
+/**
+ * Setting up the different variants we will be testing with.
+ *
+ * @note  The `MaxDiff` test have much fewer transactions than all others, this is
+ *        because each transaction is LARGE, so the block size in kb is hit.
+ *        I decided that 1/4 should be acceptable, and still small enough to work.
+ */
+const variants: TestVariant[] = [
+  new TestVariant(10, 36, TxComplexity.Deployment),
+  new TestVariant(10, 36, TxComplexity.PrivateTransfer),
+  new TestVariant(10, 36, TxComplexity.PublicTransfer),
+  new TestVariant(10, 9, TxComplexity.MaxDiff),
+];
```

Comment:
Would be nice to also control which variants are run via env vars

Reply:
How would you want to use it? The main thing I had was just if we want to build the blocks, it will take a long time, e.g., 10-30 minutes depending on the complexity. Running the sync part only is fairly quick.

Reply:
I was thinking of speeding up block building for local testing (when we have a breaking change, for instance) by just setting up a single variant.

Reply:
Not fully following. I won't add this for now, might make sense later.

Reply:
Sounds good, it's not really important.